test/extended/single_node/topology.go

```diff
@@ -0,0 +1,42 @@
+package single_node
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	exutil "github.com/openshift/origin/test/extended/util"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+	"strings"
+)
+
+var _ = Describe("[sig-arch] Cluster topology single node tests", func() {
+	oc := exutil.NewCLIWithoutNamespace("singlenode").Verbose()
+
+	It("Verify that topology API is respected", func() {
+		out, err := oc.Run("get").Args("infrastructure", "cluster",
+			"-o", "template", "--template",
+			`{{.status.controlPlaneTopology}}`,
+		).Output()
+		Expect(err).NotTo(HaveOccurred())
+
+		if out != "SingleReplica" {
+			e2eskipper.Skipf(
+				"Test is only relevant for SingleReplica control plane topology (got %s)", out)
+		}
+
+		out, err = oc.Run("get").Args("--all-namespaces", "deployment",
```

Comment:
Should we narrow this down to just the openshift namespaces?

Reply:
Probably, yeah. Will do that